### PR TITLE
improve error checking for `astro dev upgrade-check`

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

> Describe the purpose of this pull request.

## 🎟 Issue(s)

Resolves astronomer/issues#XXXX

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
practical tests:

**Airflow 1.10.12 image:**
```
$ astro-cli dev upgrade-check
Running: bash -c pip install --no-deps 'apache-airflow-upgrade-check'; python -c 'from packaging.version import Version
from airflow import __version__
if Version(__version__) < Version("1.10.13"):
  print("upgrade to 1.10.13 first");exit(1)
else:
  from airflow.upgrade.checker import __main__;__main__()'
Looking in indexes: https://pypi.org/simple, https://pip.astronomer.io/simple/
Requirement already satisfied: apache-airflow-upgrade-check in /usr/local/lib/python3.7/site-packages (1.0.0)
upgrade to 1.10.13 first
```

**Comment by andriisoldatenko** [comment]
Airflow `ap-airflow:1.10.14-1-buster-onbuild`:

```
project_4244 astro-cli dev upgrade-check
Running: bash -c pip install --no-deps 'apache-airflow-upgrade-check'; python -c 'from packaging.version import Version
from airflow import __version__
if Version(__version__) < Version("1.10.13"):
  print("upgrade to 1.10.13 first");exit(1)
else:
  from airflow.upgrade.checker import __main__;__main__()'
Looking in indexes: https://pypi.org/simple, https://pip.astronomer.io/simple/
Collecting apache-airflow-upgrade-check
  Downloading https://files.pythonhosted.org/packages/49/47/78970f60a9df509d1cc551eb135098c6b95e48ede4126b811fbacf4a86c8/apache_airflow_upgrade_check-1.0.0-py2.py3-none-any.whl (44kB)
Installing collected packages: apache-airflow-upgrade-check
Successfully installed apache-airflow-upgrade-check-1.0.0

====== STATUS ======

Check for latest versions of apache-airflow and checkerSUCCESS
Remove airflow.AirflowMacroPlugin classSUCCESS
Chain between DAG and operator not allowed.SUCCESS
Connection.conn_id is not uniqueSUCCESS
Connection.conn_type is not nullableSUCCESS
Fernet is enabled by defaultSUCCESS
GCP service account key deprecationSUCCESS
Changes in import paths of hooks, operators, sensors and othersFAIL
Users must delete deprecated configs for KubernetesExecutorFAIL
Legacy UI is deprecated by defaultSUCCESS
Logging configuration has been moved to new sectionSUCCESS
Removal of Mesos ExecutorSUCCESS
Users must set a kubernetes.pod_template_file valueFAIL
SendGrid email uses old airflow.contrib moduleSUCCESS
Changes in import path of remote task handlersSUCCESS
Jinja Template Variables cannot be undefinedSUCCESS
Found 4 problems.

= RECOMMENDATIONS ==

Changes in import paths of hooks, operators, sensors and others
---------------------------------------------------------------
Many hooks, operators and other classes has been renamed and moved. Those changes were part of unifying names and imports paths as described in AIP-21.
The `contrib` folder has been replaced by `providers` directory and packages:
https://github.com/apache/airflow#backport-packages

Problems:

  1.  Using `airflow.operators.bash_operator.BashOperator` will be replaced by `airflow.operators.bash.BashOperator`. Affected file: /usr/local/airflow/dags/example-dag.py
  2.  Using `airflow.operators.python_operator.PythonOperator` will be replaced by `airflow.operators.python.PythonOperator`. Affected file: /usr/local/airflow/dags/example-dag.py

Users must delete deprecated configs for KubernetesExecutor
-----------------------------------------------------------
In Airflow 2.0, KubernetesExecutor Users need to set a pod_template_file as a base
value for all pods launched by the KubernetesExecutor. Many Kubernetes configs are no longer
needed once this pod_template_file has been generated.


Problems:

  1.  The following invalid keys were found in your airflow.cfg:

airflow_configmap
airflow_local_settings_configmap
dags_in_image
dags_volume_subpath
dags_volume_mount_point
dags_volume_claim
logs_volume_subpath
logs_volume_claim
dags_volume_host
logs_volume_host
env_from_configmap_ref
env_from_secret_ref
git_repo
git_branch
git_sync_depth
git_subpath
git_sync_rev
git_user
git_password
git_sync_root
git_sync_dest
git_dags_folder_mount_point
git_ssh_key_secret_name
git_ssh_known_hosts_configmap_name
git_sync_credentials_secret
git_sync_container_repository
git_sync_container_tag
git_sync_init_container_name
git_sync_run_as_user
worker_service_account_name
image_pull_secrets
gcp_service_account_keys
affinity
tolerations
run_as_user
fs_group

                    Please generate a pod_template_file by running `airflow generate_pod_template`                    and delete these keys.

Users must set a kubernetes.pod_template_file value
---------------------------------------------------
In Airflow 2.0, KubernetesExecutor Users need to set a pod_template_file as a base
value for all pods launched by the KubernetesExecutor


Problems:

  1.  Please create a pod_template_file by running `airflow generate_pod_template`.
This will generate a pod using your aiflow.cfg settings


```

**Review by ashb — APPROVED** [review]

File: cmd/airflow.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func airflowUpgradeCheck(cmd *cobra.Command, args []string) error {
 	cmd.SilenceUsage = true
 
 	// Add airflow command, to simplify astro cli usage
-	args = append([]string{"bash", "-c", "pip install --no-deps 'apache-airflow-upgrade-check'; airflow upgrade_check"})
+	args = append([]string{"bash", "-c", "pip install --no-deps 'apache-airflow-upgrade-check'; python -c 'from packaging.version import Version\nfrom airflow import __version__\nif Version(__version__) < Version(\"1.10.13\"):\n  print(\"upgrade to 1.10.13 first\");exit(1)\nelse:\n  from airflow.upgrade.checker import __main__;__main__()'"})
```
Comment (kaxil) [inline cmd/airflow.go:401]:
```suggestion
	args = append([]string{"bash", "-c", "pip install --no-deps 'apache-airflow-upgrade-check'; python -c 'from packaging.version import Version\nfrom airflow import __version__\nif Version(__version__) < Version(\"1.10.14\"):\n  print(\"upgrade to 1.10.14 first\");exit(1)\nelse:\n  from airflow.upgrade.checker import __main__;__main__()'"})
```

**Review by kaxil — APPROVED** [review]

File: cmd/airflow.go [hunk 2]
```diff
@@ -398,7 +398,7 @@ func airflowUpgradeCheck(cmd *cobra.Command, args []string) error {
 	cmd.SilenceUsage = true
 
 	// Add airflow command, to simplify astro cli usage
-	args = append([]string{"bash", "-c", "pip install --no-deps 'apache-airflow-upgrade-check'; airflow upgrade_check"})
+	args = append([]string{"bash", "-c", "pip install --no-deps 'apache-airflow-upgrade-check'; python -c 'from packaging.version import Version\nfrom airflow import __version__\nif Version(__version__) < Version(\"1.10.14\"):\n  print(\"upgrade to 1.10.14 first\");exit(1)\nelse:\n  from airflow.upgrade.checker import __main__;__main__()'"})
```
Comment (schnie) [inline cmd/airflow.go:401]:
```suggestion
	args = append([]string{"bash", "-c", "pip install --no-deps 'apache-airflow-upgrade-check'; python -c 'from packaging.version import Version\nfrom airflow import __version__\nif Version(__version__) < Version(\"1.10.14\"):\n  print(\"Please upgrade your image to Airflow 1.10.14 first, then try again.\");exit(1)\nelse:\n  from airflow.upgrade.checker import __main__;__main__()'"})
```

Comment (schnie) [reply cmd/airflow.go:401]: Just expanding the message here a bit, does that make sense here?

**Review by schnie — APPROVED** [review]

**Pull request merged**